src/components/LHNOptionsList/LHNOptionsList.js

```diff
@@ -189,5 +234,8 @@ export default compose(
         draftComments: {
             key: ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT,
         },
+        transactionViolations: {
+            key: ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS,
```

Comment:
Out of curiosity, is there a reason we're taking all the transaction violations, and not setting this to a function that takes a transactionID and returns the transaction violations for a specific transaction?

Reply:
Are we able to determine the transactionId within the `withOnyx` call? I'm not aware of how we would narrow that down otherwise without doing the same computation we do later on after it's passed down to `transactionThreadHasViolations`

Reply:
@cead22 -- this I'll have to look into a bit more as this is my first contact with `withOnyx`

Reply:
Will this work?

```ts
export default compose(
    withCurrentReportID,
    withOnyx({
        // ..
        transactionViolations: {
            key: ({currentReportId}) => `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${currentReportId}`,
        },
    }),
)(LHNOptionsList);
```

Reply:
why need to filter out with currentReportId here? This is OptionsList, not OptionRow